### PR TITLE
Profitwell Integration

### PR DESCRIPTION
**Pull request by akashrpo**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/323)

### CODE REVIEW

File: integrations/ProfitWell/browser.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+          (i[o].q = i[o].q || []).push(arguments);
+        };
+      a = s.createElement(g);
+      m = s.getElementsByTagName(g)[0];
+      a.async = 1;
+      a.src =
+        r +
+        "?auth=" +
+        s.getElementById(o + "-js").getAttribute(this.publicApiToken);
+      m.parentNode.insertBefore(a, m);
```
Comment (saikumarrs) [inline integrations/ProfitWell/browser.js:29]: Wouldn't it be easier to use the ScriptLoader utility?

File: integrations/ProfitWell/browser.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      return;
+    }
+
+    if (payload.user_email) {
+      delete payload.user_id;
+    } else {
+      delete payload.user_email;
+    }
+
+    payload = removeUndefinedAndNullValues(payload);
```
Comment (saikumarrs) [inline integrations/ProfitWell/browser.js:93]: Please see if you can extract a method to re-factor "payload data validation and cleanup" which can be used in multiple places

File: integrations/ProfitWell/browser.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      user_id: get(message, "context.userId"),
+    };
+
+    if (!payload.user_email && !payload.user_id) {
+      logger.error("User parameter (email or id) is required");
+      return;
+    }
+
+    if (payload.user_email) {
+      delete payload.user_id;
```
Comment (arajguha) [inline integrations/ProfitWell/browser.js:88]: can we avoid using the `delete` operation?

File: integrations/ProfitWell/browser.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      m.parentNode.insertBefore(a, m);
+    })(
+      window,
+      document,
+      "profitwell",
+      "script",
+      "https://public.profitwell.com/js/profitwell.js"
+    );
+
+    const userId = generateUUID().toString();
```
Comment (arajguha) [inline integrations/ProfitWell/browser.js:43]: lets put this uuid generation inside the if block

File: integrations/ProfitWell/browser.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      m.parentNode.insertBefore(a, m);
+    })(
+      window,
+      document,
+      "profitwell",
+      "script",
+      "https://public.profitwell.com/js/profitwell.js"
+    );
+
+    if (this.siteType === "marketing") {
```
Comment (arajguha) [inline integrations/ProfitWell/browser.js:43]:
```suggestion
    if (this.siteType === "marketing") {
      window.profitwell("start", { user_id: generateUUID().toString(); });
      return;
    }
```

**Pull request merged**